backend/scripts/install_db.py

```
@@ -357,6 +358,56 @@ def run_server_sql(
+    直到运行期才在个别页面炸出 SQL 语法/类型错误，最难排查。下限依据（本仓实测）：
+    MySQL 5.7.8 起才有 json 类型（基线 58 个 json 列）与虚拟生成列上的唯一索引；
+    MariaDB 取 Laravel 12 官方支持矩阵的 10.3。版本串解析不出来时保守放行。
+    """
+    match = re.search(r"(\d+\.\d+(?:\.\d+)?)", raw_version)
+    if not match:
+        return
+    version = tuple(int(part) for part in match.group(1).split("."))
+
+    if "mariadb" in raw_version.lower():
+        if version < (10, 3):
+            fail(f"MariaDB 版本过低（当前 {raw_version}）：最低需要 10.3。")
```

Ruff (0.16.2)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)